runtime/nvqir/CircuitSimulator.h

```diff
@@ -616,6 +618,10 @@ class CircuitSimulatorBase : public CircuitSimulator {
       addQubitToState();
   }
 
+  /// @brief Add (appending) the given simulation state to the current simulator
+  /// state.
+  virtual void addQubitsToState(const cudaq::SimulationState &state) = 0;
```

Comment:
Random thought: could this be a NOP default implementation instead of `= 0` here? That way the derived class wouldn't have to define this method if for example it didn't support initial states for some reason.

Reply:
Sure, we can do that. I made it `=0` mainly for my benefit to know where I need to implement it. 
I'll update the code.

Reply:
Either way is ok by me.